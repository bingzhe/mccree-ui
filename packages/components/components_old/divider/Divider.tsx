import * as React from "react";
import classNames from "classnames";
import { ConfigContext } from "../config-provider";
import useSetColor from "../hooks/useSetColor";

export interface DividerProps extends React.HtmlHTMLAttributes<HTMLDivElement> {
    type?: "horizontal" | "vertical";
    align?: "left" | "right" | "center";
    color?: string;
    dashed?: boolean;
    plain?: boolean;
}

const dividerProps = {
    type: "horizontal" as "horizontal" | "vertical",
    align: "center" as "left" | "right" | "center"
};

const Divider: React.FC<DividerProps> = (props) => {
    const { className, type, align, color, dashed, plain, children, ...restProps } = props;

    const dividerRef = React.useRef<HTMLDivElement>(null);
    const { getPrefixCls } = React.useContext(ConfigContext);
    const prefixCls = getPrefixCls("divider");

    const classes = classNames(prefixCls, className, `${prefixCls}-${type}`, {
        [`${prefixCls}-text`]: !!children,
        [`${prefixCls}-text-${align}`]: !!children,
        [`${prefixCls}-plain`]: !!plain,
        [`${prefixCls}-dashed`]: !!dashed
    });

    useSetColor(dividerRef, "border-color", color, true);
    useSetColor(dividerRef, "text-color", color);

    return (
        <div className={classes} {...restProps} ref={dividerRef}>
            {children && <span className={`${prefixCls}-inner-text`}>{children}</span>}
        </div>
    );
};

Divider.displayName = "MR_Divider";
Divider.defaultProps = dividerProps;

export default Divider;
