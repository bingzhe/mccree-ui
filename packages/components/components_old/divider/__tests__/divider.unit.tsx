import * as React from "react";
import { render, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Divider from "../index";

describe("Divider", () => {
    it("should render correctly", () => {
        const wrapper = render(<Divider />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should work with align", () => {
        const wrapper = render(
            <>
                <Divider align="left">divider</Divider>
                <Divider align="right">divider</Divider>
                <Divider align="center">divider</Divider>
            </>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should work with type", () => {
        const wrapper = render(
            <>
                <Divider type="horizontal">divider</Divider>
                <Divider type="vertical" />
            </>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should work with color", () => {
        const wrapper = render(
            <>
                <Divider color="primary">divider</Divider>
                <Divider color="secondary">divider</Divider>
                <Divider color="success">divider</Divider>
                <Divider color="warning">divider</Divider>
                <Divider color="error">divider</Divider>
                <Divider color="info">divider</Divider>
            </>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should work with RGB HEX", () => {
        const wrapper = render(
            <>
                <Divider color="rgba(172,17,236,1)">divider</Divider>
                <Divider color="#1ED39A">divider</Divider>
            </>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should find mccree-divider classnames", () => {
        const wrapper = mount(<Divider color="rgba(172,17,236,1)">divider</Divider>);
        expect(wrapper.find(".mccree-divider").length).toBe(1);
    });
});
