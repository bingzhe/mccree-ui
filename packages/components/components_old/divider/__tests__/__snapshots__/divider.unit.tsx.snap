// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render correctly 1`] = `
<div
  class="mccree-divider mccree-divider-horizontal"
/>
`;

exports[`Divider should work with RGB HEX 1`] = `
Array [
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
]
`;

exports[`Divider should work with align 1`] = `
Array [
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-left"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-right"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
]
`;

exports[`Divider should work with color 1`] = `
Array [
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
]
`;

exports[`Divider should work with type 1`] = `
Array [
  <div
    class="mccree-divider mccree-divider-horizontal mccree-divider-text mccree-divider-text-center"
  >
    <span
      class="mccree-divider-inner-text"
    >
      divider
    </span>
  </div>,
  <div
    class="mccree-divider mccree-divider-vertical"
  />,
]
`;
